src/rcpp_octave.cpp

```diff
@@ -133,22 +147,35 @@ bool octave_session(bool start=true, bool with_warnings = true, bool verbose = f
 
 		// instantiate the Octave interpreter
 		int narg = 4;
-		string_vector cmd_args(narg);
+		int narg_overflow = optind + 1; // workaround for bug in Octave 4.2 and later:
+		// octave.cc (v4.2, 4.3+) cmdline_options::cmdline_options() uses
+		// argv + octave_optind_wrapper (), where argv is the char** produced below
+		// and optind is the number of arguments getopt.h reports for the executable,
+		// having nothing to do with the args produced here. If the executable
+		// was called with more than narg arguments, this gives a SEGFAULT.
+		string_vector cmd_args(narg + narg_overflow);
 		cmd_args(0) = std::string("RcppOctave");
 		cmd_args(1) = std::string("--quiet");
 		// Try avoid corruption of R console with Octave console outputs
 		// [suggested by Albert Graef]
 		cmd_args(2) = std::string("--no-line-editing");
 		cmd_args(3) = std::string("--no-history");
+		//static octave::cmdline_options opts(narg, cmd_args.c_str_vec());
+		octave::cmdline_options opts(narg, cmd_args.c_str_vec());
```

Comment:
Why create a patched up `cmdline_options` object here and not just construct the application below by passing nargs and argv to the constructor directly? In that case octave would create the `cmdline_options` on its end and trigger the SEGFAULT.

The problematic piece of code referred to in the comment at lines 150-155 is in `libinterp/octave.cc cmdline_options::cmdline_options()`
```
    ...
    m_remaining_args = string_vector (argv+octave_optind_wrapper (),
                                      argc-octave_optind_wrapper ());
```